packages/angular/build/src/tools/vite/middlewares/index-html-middleware.ts

```diff
@@ -28,13 +29,14 @@ export function createAngularIndexHtmlMiddleware(
     // Parse the incoming request.
     // The base of the URL is unused but required to parse the URL.
     const pathname = pathnameWithoutBasePath(req.url, server.config.base);
-    if (pathname !== '/' && pathname !== '/index.html') {
```

Comment:
This part is already handled in https://github.com/angular/angular-cli/blob/main/packages/angular/build/src/tools/vite/middlewares/html-fallback-middleware.ts